File: web/src/main/java/com/marklogic/hub/web/model/JobModel.java
package com.marklogic.hub.web.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;
import com.fasterxml.jackson.databind.JsonNode;
import java.util.List;

@JsonPropertyOrder({"id", "flowId", "flowName", "user", "status", "successfulEvents", "failedEvents", "startTime", "endTime", "lastAttemptedStep", "lastCompletedStep"})
public class JobModel {
    public String id;
    public String flowId;
    public String flowName;
    public String user;
    public String status;
    public long successfulEvents;
    public long failedEvents;
    public String startTime;
    public String endTime;
    public String lastAttemptedStep;
    public String lastCompletedStep;

    public static class JobStep {
        public int stepNumber;
        public String id;
        public String name;
        public String stepDefinitionName;
        public String stepDefinitionType;
        public String targetDatabase;
        public String targetEntity;
        public String status;
        public long totalEvents;
        public long successfulEvents;
        public long failedEvents;
        public long successfulBatches;
        public long failedBatches;
        public boolean success;
        public String startTime;
        public String endTime;
        public JsonNode stepOutput;
        public String fullOutput;
    }

    @JsonProperty("steps")
    public List<JobStep> stepModels;
}

File: web/src/main/ui/app/components/jobs-new/models/job.model.ts
import * as _ from "lodash";
// export enum StepType {
//   INGESTION = 'INGESTION',
//   MAPPING = 'MAPPING',
//   MASTERING = 'MASTERING',
//   CUSTOM = 'CUSTOM'
// }

export class Job {
  public id: string;
  public flowName: string;
  public flowId: string;
  public user: string;
  public status: string;
  public startTime: string;
  public endTime: string;
  public successfulEvents: number;
  public failedEvents: number;
  public lastAttemptedStep: string;
  public lastCompletedStep: string;
  public steps: Array<any> = [];

  private constructor() {}

  static fromJSON(json) {

    const result = new Job();

    if (json.id) {
      result.id = json.id;
    }
    if (json.flowName) {
      result.flowName = json.flowName;
    }
    if (json.flowId) {
      result.flowId = json.flowId;
    }
    if (json.user) {
      result.user = json.user;
    }
    if (json.status) {
      result.status = json.status;
    }
    if (json.startTime) {
      result.startTime = json.startTime;
    }
    if (json.endTime) {
      result.endTime = json.endTime;
    }
    if (!_.isUndefined(json.successfulEvents)) {
      result.successfulEvents = json.successfulEvents;
    }
    if (!_.isUndefined(json.failedEvents)) {
      result.failedEvents = json.failedEvents;
    }
    if (json.lastAttemptedStep) {
      result.lastAttemptedStep = json.lastAttemptedStep;
    }
    if (json.lastCompletedStep) {
      result.lastCompletedStep = json.lastCompletedStep;
    }
    if (json.steps) {
      result.steps = json.steps;
    }
    return result;
  }
}
